### ISSUE TITLE
Optional minimal file retention mode

### ISSUE THREAD

**Issue by NGoetz** [smash-transport/smash-vhlle-hybrid#39]
Users should have the option to only preserve/generate the minimum amount of intermediate files. By default we keep a lot of files, in order to keep track of the process, but an experienced user should be able to use a more lustre friendly execution mode.

**Comment by AxelKrypton** [comment]
I am not sure it is a good idea or at least that it is worth the effort. 😅 How many files do we store that you would get rid of in this new mode? Could you make a folder tree as example and somehow mark what is kind of superfluous? 🤔

**Comment by NGoetz** [comment]
Everything except the final particle list from the Afterburner is eventually superflous.
The user cannot at the moment remove files once unneeded, as the workflow is one continous process.

**Comment by AxelKrypton** [comment]
> Everything except the final particle list from the Afterburner is eventually superflous.

I believe this is way more intricate that it might sound. We offer full flexibility, so we cannot even assume the `Afterburner` is the last stage run by the user. We could define the "result" for each stage and only keep that for the last stage run.

> The user cannot at the moment remove files once unneeded, as the workflow is one continous process.

This sentence makes me guess that you would like this feature to happen on the way internally to the sampler. Something like: Once I set-up and I am about running a given phase, I can delete everything about previous phases that is not input for what I am about to run. And once I did the last one, I remove everything but the result. Would this still be fine in case of failure? What about reproducibility? In that mode we would need to store the full hybrid handler config file as part of "the result". 🤔

**Comment by NGoetz** [comment]
> Would this still be fine in case of failure? What about reproducibility? In that mode we would need to store the full hybrid handler config file as part of "the result".

I agree. In case of failure, we usually know that the failure happened in the phase currently running, as this setup is anyway rather for well-tested production. If a stage successfully finished, the user should have the functionality to remove its files.

**Comment by RenHirayama** [comment]
I would also bring up that since SMASH copies the config file to the data folder, we could suppress the Afterburner/IC copies of the config

**Comment by AxelKrypton** [comment]
Just to keep track of a possible alternative approach.

Cleaning up sounds to me like a possibly disentangled task from the production run. We could add a new `clean` execution mode to the handler which removes _whatever we decide it makes sense to be removed_. This as mode to be used once the production run is completed. One should clarify details (like whether a hybrid handler config file is needed for this mode and in case what should contain) when dealing with this issue. Then one could also implement a `--live` command line option such that the cleaning is done "waiting for folders". Normally the clean up would be a loop over stages deleting files into the folders, with the `--live` option the handler would wait for the second stage folder to exist to clean up the first. This would be meant to be executed on clusters in background in parallel to the production run.

An alternative/complementary feature might be to add a `archive` execution mode, which makes the handler copy "the result" somewhere. However, this is harder than it might sound, because "the result" should be defined for all possible final stages.

### Side remark

I would not make the handler, by default, delete files, but just move them into a ***Trash*** folder. The user could then delete such a folder by hand and, yes, it would be their responsibility to do this extra step, but it would be a rescue in case of distraction. It is always possible to make the handler remove the folder using a `--delete` option.
